File: Filepath Existence/Filepath Existence.py
#
#   Filepath Existence Checker, Python
#   Module written by Brandon Arvanaghi
#   Website: arvanaghi.com 
#   Twitter: @arvanaghi
#

import os.path

EvidenceOfSandbox = []

FilePathsToCheck = [r'C:\windows\System32\Drivers\Vmmouse.sys', 
r'C:\windows\System32\Drivers\vm3dgl.dll', r'C:\windows\System32\Drivers\vmdum.dll', 
r'C:\windows\System32\Drivers\vm3dver.dll', r'C:\windows\System32\Drivers\vmtray.dll',
r'C:\windows\System32\Drivers\vmci.sys', r'C:\windows\System32\Drivers\vmusbmouse.sys',
r'C:\windows\system32\Drivers\vmx_svga.sys', r'C:\windows\system32\Drivers\vmxnet.sys',
r'C:\windows\System32\Drivers\VMToolsHook.dll', r'C:\windows\System32\Drivers\vmhgfs.dll',
r'C:\windows\System32\Drivers\vmmousever.dll', r'C:\windows\System32\Drivers\vmGuestLib.dll',
r'C:\windows\System32\Drivers\VmGuestLibJava.dll', r'C:\windows\System32\Drivers\vmscsi.sys',
r'C:\windows\System32\Drivers\VBoxMouse.sys', r'C:\windows\System32\Drivers\VBoxGuest.sys',
r'C:\windows\System32\Drivers\VBoxSF.sys', r'C:\windows\System32\Drivers\VBoxVideo.sys',
r'C:\windows\System32\vboxdisp.dll', r'C:\windows\System32\vboxhook.dll',
r'C:\windows\System32\vboxmrxnp.dll', r'C:\windows\System32\vboxogl.dll',
r'C:\windows\System32\vboxoglarrayspu.dll', r'C:\windows\System32\vboxoglcrutil.dll',
r'C:\windows\System32\vboxoglerrorspu.dll', r'C:\windows\System32\vboxoglfeedbackspu.dll',
r'C:\windows\System32\vboxoglpackspu.dll', r'C:\windows\System32\vboxoglpassthroughspu.dll',
r'C:\windows\System32\vboxservice.exe', r'C:\windows\System32\vboxtray.exe',
r'C:\windows\System32\VBoxControl.exe']

for FilePath in FilePathsToCheck:
	print(FilePath)
	if os.path.isfile(FilePath):
		EvidenceOfSandbox.append(FilePath)

if not EvidenceOfSandbox:
	print("Proceed!")
else:
	print(EvidenceOfSandbox)

File: Filepath Existence/filepath_existence.pl
#
#   Filepath existence checker, Perl
#   Module written by Brandon Arvanaghi 
#   Website: arvanaghi.com 
#   Twitter: @arvanaghi
#

my @EvidenceOfSandbox;

my @FilePathsToCheck = (q{C:\Windows\Sysnative\Drivers\vmmouse.sys}, 
q{C:\Windows\Sysnative\Drivers\vm3dgl.dll}, q{C:\Windows\Sysnative\Drivers\vmdum.dll}, 
q{C:\Windows\Sysnative\Drivers\vm3dver.dll}, q{C:\Windows\Sysnative\Drivers\vmtray.dll},
q{C:\Windows\Sysnative\Drivers\vmci.sys}, q{C:\Windows\Sysnative\Drivers\vmusbmouse.sys},
q{C:\Windows\Sysnative\Drivers\vmx_svga.sys}, q{C:\Windows\Sysnative\Drivers\vmxnet.sys},
q{C:\Windows\Sysnative\Drivers\VMToolsHook.dll}, q{C:\Windows\Sysnative\Drivers\vmhgfs.dll},
q{C:\Windows\Sysnative\Drivers\vmmousever.dll}, q{C:\Windows\Sysnative\Drivers\vmGuestLib.dll},
q{C:\Windows\Sysnative\Drivers\VmGuestLibJava.dll}, q{C:\Windows\Sysnative\Drivers\vmscsi.sys},
q{C:\Windows\Sysnative\Drivers\VBoxMouse.sys}, q{C:\Windows\Sysnative\Drivers\VBoxGuest.sys},
q{C:\Windows\Sysnative\Drivers\VBoxSF.sys}, q{C:\Windows\Sysnative\Drivers\VBoxVideo.sys},
q{C:\Windows\Sysnative\vboxdisp.dll}, q{C:\Windows\Sysnative\vboxhook.dll},
q{C:\Windows\Sysnative\vboxmrxnp.dll}, q{C:\Windows\Sysnative\vboxogl.dll},
q{C:\Windows\Sysnative\vboxoglarrayspu.dll}, q{C:\Windows\Sysnative\vboxoglcrutil.dll},
q{C:\Windows\Sysnative\vboxoglerrorspu.dll}, q{C:\Windows\Sysnative\vboxoglfeedbackspu.dll},
q{C:\Windows\Sysnative\vboxoglpackspu.dll}, q{C:\Windows\Sysnative\vboxoglpassthroughspu.dll},
q{C:\Windows\Sysnative\vboxservice.exe}, q{C:\Windows\Sysnative\vboxtray.exe},
q{C:\Windows\Sysnative\VBoxControl.exe});

foreach $FilePath (@FilePathsToCheck) {
	if (-e $FilePath) {
		push(@EvidenceOfSandbox, $FilePath);
	}
}

if (!@EvidenceOfSandbox) {
	print "No files exist on disk that suggest we are running in a sandbox. Proceed!\n";
} else {
	print "The following files on disk suggest we are running in a sandbox. Do not proceed.\n";
	print join(", ", @EvidenceOfSandbox), "\n";
}
